azure/compute/src/main/java/org/jclouds/azurecompute/compute/strategy/AzureComputeServiceAdapter.java

```diff
@@ -0,0 +1,326 @@
+            parameters.setConfigurationSets(configlist);
+
+            client.getVirtualMachineApi().addRole(group, hostedService.getDeployments().get(0).getName(), parameters);
+        }
+
+        VirtualMachine virtualMachine = getNode(name);
+        virtualMachine.setServiceName(group);
+        return new NodeAndInitialCredentials<VirtualMachine>(virtualMachine, name, null);
+    }
+
+    @Override
+    public Iterable<RoleSizeListResponse.RoleSize> listHardwareProfiles() {
+        logger.info("listHardwareProfiles");
+        return client.getRoleSizeApi().listRoleSizes();
+//        return AzureHardwareSupplier.H_ALL;
+
+    }
+
+    @Override
+    public Iterable<OSImage> listImages() {
+        return Collections2.transform(client.getImageApi().list(), osImageListResponseToOSImage);
+    }
+
+    @Override
+    public Iterable<VirtualMachine> listNodes() {
+        logger.info("listNodes");
+        List<VirtualMachine> virtualMachines = new ArrayList<VirtualMachine>();
+        List<HostedServiceListResponse.HostedService> hostedServices = client.getHostedServiceApi().list();
+        for (HostedServiceListResponse.HostedService hostedService : hostedServices) {
+            HostedServiceGetDetailedResponse serviceResponse = client.getHostedServiceApi().getHostedServiceWithDeployments(hostedService.getServiceName());
+            if (!serviceResponse.getDeployments().isEmpty()) {
+                HostedServiceGetDetailedResponse.Deployment deployment = serviceResponse.getDeployments().get(0);
+                for (Role role : deployment.getRoles()) {
+                    VirtualMachine virtualMachine = serviceResponseToVirtualMachine.apply(serviceResponse);
+                    virtualMachine.setRole(role);
+                    virtualMachines.add(virtualMachine);
+                }
+            }
+//            addDeploymentIfExists(deployments, hostedService.getServiceName(), DeploymentSlot.Staging);
+//            addDeploymentIfExists(deployments, hostedService.getServiceName(), DeploymentSlot.Production);
+        }
+        return virtualMachines;
+    }
+
+    private void addDeploymentIfExists(List<VirtualMachine> virtualMachines, String serviceName, DeploymentSlot deploymentSlot) {
+        try {
+            DeploymentGetResponse deploymentResponse =
+                    client.getVirtualMachineApi().getDeploymentBySlot(serviceName, deploymentSlot);
+            if (deploymentResponse != null) {
+                virtualMachines.add(deploymentResponseToVirtualMachine.apply(deploymentResponse));
+            }
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    @Override
+    public Iterable<VirtualMachine> listNodesByIds(final Iterable<String> ids) {
+        Set<Long> idsAsLongs = FluentIterable.from(ids)
+                .transform(toLong())
+                .toSet();
+        logger.info("listNodesByIds()");
+        throw new UnsupportedOperationException("Should not be here");
```

Comment:
This must be implemented. If there is not an API call to get N nodes by id, then just call the existing `listNodes()` method and filter afterwards.